### PR TITLE
School library associations

### PR DESCRIPTION
**Pull request by ibugithub**
Description:
📚 Introducing a comprehensive library management system with classes for students, classrooms, books, and rentals. Manage student enrollment, book rentals, and data effortlessly using object-oriented programming principles.

Changes Made:
1. 🏫 Implemented a `Classroom` class for student enrollment management.
2. 🎓 Created a `Student` class (extends `Person`) for student-specific attributes and behaviors.
3. 📖 Developed a `Book` class for easy book title and author management.
4. 📚 Designed a `Rental` class to efficiently track book rentals and associated persons.

Features:
- 🏫 Manage student enrollments with the `Classroom` class.
- 🎓 Specialized behaviors for students in the `Student` class.
- 📖 Organize book titles and authors with the `Book` class.
- 📚 Efficiently track book rentals and persons with the `Rental` class.

Checklist:
- ✔️ Resolved linting issues and adhered to coding best practices.
- ✔️ Followed Gitflow workflow for version control and branching.
- ✔️ Thoroughly documented implementation details for each class.

Additional Notes:
- ℹ️ Demonstrates proper application of object-oriented design principles.
- ✔️ Tested and verified functionality of all implemented classes.
- 📂 Organized codebase into separate files for readability.
- 🚀 Enhances management of students, classrooms, books, and rentals.

### CODE REVIEW

File: class_room.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  attr_reader :students
+
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  def add_student(student)
+    @students << student
+  end
```
Comment (TedLivist) [inline class_room.rb]: - [ ] We also need to associate the classroom with the passed student.

  ```
  def add_student(student)
    @students << student
    student.classroom = self
  end
  ```

File: student_class.rb [hunk 2]
```diff
@@ -1,15 +1,13 @@
-require './personClass'
+require './person_class'
 
 class Student < Person
-  def initialize(age, classroom, name = 'Unknown', parent_permission: true)
-    super(age, name, parent_permission)
+  def initialize(age, classroom, name = 'Unknown')
+    super(age, name)
     @classroom = classroom
+    classroom.add_student(self)
```
Comment (TedLivist) [inline student_class.rb:9]: - [ ] The Student class should have its own method for associating with the classroom also.

  ```
  def classroom=(classroom)
    @classroom = classroom
    classroom.students.push(self) unless classroom.students.include?(self)
  end
  ```

File: student_class.rb [hunk 3]
```diff
@@ -1,15 +1,13 @@
-require './personClass'
+require './person_class'
 
 class Student < Person
```
Comment (TedLivist) [inline student_class.rb:3]: - [ ] The classroom variable should have a getter. Kindly add that.

  ```
  attr_reader :classroom
  ```

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @ibugithub,

Good job so far!
✔️ Implementation of getters and setters
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by OLIPLICHE — APPROVED** [review]
Hi @ibugithub, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights
- Applied the changes requested from the previous review (✅)) 👏
- Created associations between classes 👏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**